gitops/snapshot.go

```diff
@@ -342,6 +351,28 @@ func MarkSnapshotAsFailed(ctx context.Context, adapterClient client.Client, snap
 	return nil
 }
 
+// MarkSnapshotAsCanceled updates the AppStudio Test canceled condition for the Snapshot to canceled.
+// If the patch command fails, an error will be returned.
+func MarkSnapshotAsCanceled(ctx context.Context, adapterClient client.Client, snapshot *applicationapiv1alpha1.Snapshot, message string) error {
+	patch := client.MergeFrom(snapshot.DeepCopy())
+	condition := metav1.Condition{
+		Type:    AppStudioTestCanceledCondition,
```

Comment:
I'm thinking it might be better to use the existing `AppStudioIntegrationStatusCondition` here instead of creating a new condition, wdyt?

Reply:
Changed to the existing condition.